### ISSUE TITLE
Plugin tries to wrap PropTypes.shape

### ISSUE THREAD

**Issue by mmakarin** [gaearon/react-transform-catch-errors#25]
Source:

```
import React, { PropTypes } from "react";
import { PureComponent } from "react-pure-render";

export default class Table extends PureComponent {
    static propTypes = {
        columns: PropTypes.arrayOf(PropTypes.shape({
            title: PropTypes.string,
            render: PropTypes.func
        })),
        data: PropTypes.array
    };

    ...
```

Output

```
var Table = _wrapComponent("Table")((_temp = _class = function (_PureComponent) {
    _inherits(Table, _PureComponent);

    function Table() {
        _classCallCheck(this, Table);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(Table).apply(this, arguments));
    }

    _createClass(Table, [{
        key: "render",
        value: function render() {
             ...
        }
    }]);

    return Table;
}(_reactPureRender.PureComponent), _class.propTypes = {
    columns: _react2.PropTypes.arrayOf(_wrapComponent("_component")(_react2.PropTypes.shape({
        title: _react2.PropTypes.string,
        render: _react2.PropTypes.func
    }))),
    data: _react2.PropTypes.array
}, _temp));
```

And it fails on the line

```
var originalRender = ReactClass.prototype.render;
```

with the following error because PropTypes.Shape.prototype.render is undefined

```
Uncaught TypeError: Cannot read property 'render' of undefined
wrapToCatchErrors @ index.js:30(anonymous function)
@ Table.js:65(anonymous function)
@ Table.js:151(anonymous function) 
...
```

**Comment by mmakarin** [comment]
https://github.com/gaearon/babel-plugin-react-transform/issues/26

**Comment by gaearon** [comment]
This doesn’t appear to have anything to do with https://github.com/gaearon/babel-plugin-react-transform/issues/26.

https://github.com/gaearon/babel-plugin-react-transform/issues/26 is a feature request for wrapping exports.

What you see is a bug in Babel plugin where it wraps `React.whatever` instead of strictly `React.createClass`. I filed it as https://github.com/gaearon/babel-plugin-react-transform/issues/84.

A test and a fix are welcome!

**Comment by gaearon** [comment]
I think I know what the problem is.
Fix is welcome: https://github.com/gaearon/babel-plugin-react-transform/issues/84#issuecomment-185472925.

**Comment by gaearon** [comment]
Should be fixed in 2.0.1.
